lib/sequencer/src/execution/block_context_provider.rs

```diff
@@ -56,8 +59,10 @@ impl<Mempool: L2TransactionPool> BlockContextProvider<Mempool> {
         node_version: semver::Version,
         genesis: Arc<Genesis>,
         fee_collector_address: Address,
-        base_fee_override: Option<u64>,
-        pubdata_price_override: Option<u64>,
+        base_fee_override: Option<u128>,
+        pubdata_price_override: Option<u128>,
+        pubdata_price_provider: Arc<dyn PubdataPriceProvider>,
```

Comment:
What if we replace `Arc<dyn PubdataPriceProvider>` with [`watch::Receiver<Option<PubdataPrice>>`](https://docs.rs/tokio/latest/tokio/sync/watch/struct.Receiver.html)?

This way we:
1. won't need an interface for `PubdataPriceProvider` (which doesn't seem an abstraction we're confident in)
2. can turn `GasAdjuster` into a completely detached actor, and for ENs just provide a receiver that is never updated (always has `None`)